app/src/main/java/com/github/ybecker/epforuml/chat/RealChatFragment.kt

```diff
@@ -0,0 +1,153 @@
+package com.github.ybecker.epforuml.chat
+
+import android.content.Intent
+import android.opengl.Visibility
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import com.github.ybecker.epforuml.MainActivity
+import com.github.ybecker.epforuml.R
+import com.github.ybecker.epforuml.R.*
+import com.github.ybecker.epforuml.database.DatabaseManager.db
+import com.github.ybecker.epforuml.database.DatabaseManager.user
+import com.github.ybecker.epforuml.database.Model
+import java.time.LocalDateTime
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import kotlin.collections.ArrayList
+
+/**
+ * A fragment representing a chat which is a list of messages.
+ */
+class RealChatFragment : Fragment() {
```

Comment:
I would have implemented it as an Activity, to simplify navigation and other features.